FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestKeyIDStorage.m

```diff
@@ -36,18 +47,46 @@ - (instancetype)initWithAppName:(NSString *)appName appID:(NSString *)appID {
   if (self) {
     _appName = [appName copy];
     _appID = [appID copy];
+    _userDefaults = [[NSUserDefaults alloc] initWithSuiteName:kFIRAppAttestKeyIDStorageDefaultsSuiteName];
   }
   return self;
 }
 
+- (nonnull FBLPromise<NSString *> *)setAppAttestKeyID:(nullable NSString *)keyID {
+  [self storeAppAttestKeyID:keyID];
+  return [FBLPromise resolvedWith:keyID];
+}
+
 - (nonnull FBLPromise<NSString *> *)getAppAttestKeyID {
-  // TODO: Implement.
-  return [FBLPromise resolvedWith:nil];
+  NSString *appAttestKeyID = [self appAttestKeyIDFromStorage];
+  if (appAttestKeyID) {
+    return [FBLPromise resolvedWith:appAttestKeyID];
+  } else {
+    NSError *error = [FIRAppCheckErrorUtil appAttestKeyIDNotFound];
+    FBLPromise *rejectedPromise = [FBLPromise pendingPromise];
+    [rejectedPromise reject:error];
+    return rejectedPromise;
+  }
 }
 
-- (nonnull FBLPromise<NSString *> *)setAppAttestKeyID:(nullable NSString *)keyID {
-  // TODO: Implement.
-  return [FBLPromise resolvedWith:nil];
+#pragma mark - Helpers
+
+- (void)storeAppAttestKeyID:(nullable NSString *)keyID {
+  @synchronized (self.userDefaults) {
+    if (keyID) {
+      [self.userDefaults setObject:keyID forKey:kFIRAppAttestKeyIDKey];
+    } else {
+      [self.userDefaults removeObjectForKey:kFIRAppAttestKeyIDKey];
+    }
+  }
+}
+
+- (nullable NSString *)appAttestKeyIDFromStorage {
+  NSString *appAttestKeyID = nil;
+  @synchronized (self.userDefaults) {
+    appAttestKeyID = [self.userDefaults objectForKey:kFIRAppAttestKeyIDKey];
```

Comment:
We need to store an App Attest key ID per Firebase app instance to make sure they can operate independently. The app instance is identified by the app name. In addition we need to make sure that if Firebase config changes, the app can still work correctly on user devices after update. For this we need to keep track of the app ID. The easiest way to achieve it in the code I can see is to make the app name and the app ID a part of the key. We use this approach already in the [core part of the SDK](https://github.com/firebase/firebase-ios-sdk/blob/638f5c7cf0628216af28cca96c7c57902217149b/FirebaseAppCheck/Sources/Core/Storage/FIRAppCheckStorage.m#L106) .

Reply:
Done and added a test like `testSetTokenPerApp ` seen [here](https://github.com/firebase/firebase-ios-sdk/blob/638f5c7cf0628216af28cca96c7c57902217149b/FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckStorageTests.m#L76).